src/client/actions/GuildScheduledEventUserAdd.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const Action = require('./Action');
+const { Events } = require('../../util/Constants');
+
+class GuildScheduledEventUserAddAction extends Action {
+  handle(data) {
+    const client = this.client;
+    const guild = client.guilds.cache.get(data.guild_id);
+
+    if (guild) {
+      const guildScheduledEvent = guild.scheduledEvents.cache.get(data.guild_scheduled_event_id);
+      const user = client.users.cache.get(data.user_id);
```

Comment:
yeah, was thinking about the same thing. These two are experimental events, so we can either not support them right now or only emit them when both `GuildScheduledEvent` and `User` are cached.

This could be fixed if Discord returns full objects instead of just ids for them.